clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java

```diff
@@ -106,79 +107,78 @@ public void testClientMode() throws Exception {
+        sslEngineBuilder = sslFactory.sslEngineBuilder();
 
         // Verify that context is recreated after validation on reconfigure() if config is not changed, but keystore file was modified
         keyStoreFile.setLastModified(System.currentTimeMillis() + 15000);
         sslFactory.validateReconfiguration(sslConfig);
         sslFactory.reconfigure(sslConfig);
-        assertNotSame("SSL context not recreated", sslContext, sslFactory.sslContext());
-        sslContext = sslFactory.sslContext();
+        assertNotSame("SslEngineBuilder not recreated",
+                sslEngineBuilder, sslFactory.sslEngineBuilder());
+        sslEngineBuilder = sslFactory.sslEngineBuilder();
 
         // Verify that the context is not recreated if modification time cannot be determined
         keyStoreFile.setLastModified(System.currentTimeMillis() + 20000);
         Files.delete(keyStoreFile.toPath());
         sslFactory.reconfigure(sslConfig);
-        assertSame("SSL context recreated unnecessarily", sslContext, sslFactory.sslContext());
+        assertSame("SslEngineBuilder recreated unnecessarily",
+                sslEngineBuilder, sslFactory.sslEngineBuilder());
     }
 
     @Test
     public void testReconfigurationWithoutTruststore() throws Exception {
         File trustStoreFile = File.createTempFile("truststore", ".jks");
-        Map<String, Object> sslConfig = TestSslUtils
-            .createSslConfig(false, true, Mode.SERVER, trustStoreFile, "server");
+        Map<String, Object> sslConfig = TestSslUtils.
+            createSslConfig(false, true, Mode.SERVER, trustStoreFile, "server");
         sslConfig.remove(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG);
         sslConfig.remove(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG);
         sslConfig.remove(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG);
         SslFactory sslFactory = new SslFactory(Mode.SERVER);
         sslFactory.configure(sslConfig);
-        SSLContext sslContext = sslFactory.sslContext();
+        SSLContext sslContext = sslFactory.sslEngineBuilder().sslContext();
         assertNotNull("SSL context not created", sslContext);
-        assertSame("SSL context recreated unnecessarily", sslContext, sslFactory.sslContext());
+        assertSame("SSL context recreated unnecessarily", sslContext,
+                sslFactory.sslEngineBuilder().sslContext());
         assertFalse(sslFactory.createSslEngine("localhost", 0).getUseClientMode());
 
-        trustStoreFile = File.createTempFile("truststore", ".jks");
-        sslConfig = TestSslUtils.createSslConfig(false, true, Mode.SERVER, trustStoreFile, "server");
-        sslConfig.remove(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG);
-        sslConfig.remove(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG);
-        sslConfig.remove(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG);
-        sslFactory.reconfigure(sslConfig);
-        assertNotSame("SSL context not recreated", sslContext, sslFactory.sslContext());
```

Comment:
Why was this removed? This is testing reconfiguration without a truststore.

Reply:
this is another example of the validate / reconfigure split: reconfigure no longer re-performs all the work validation does